Sources/Bookmarks/BookmarkUtils.swift

```diff
@@ -156,6 +156,16 @@ public struct BookmarkUtils {
         return (try? context.fetch(request)) ?? []
     }
 
+    public static func fetchModifiedBookmarksTitles(before date: Date, in context: NSManagedObjectContext) -> [String] {
```

Comment:
I'm going for a context-specific API, i.e. fetching titles because we want titles, not entire objects. For credentials, it spares us decrypting passwords.